xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -1701,6 +1735,43 @@ bool CGUIDialogVideoInfo::ManageVideoItemArtwork(const std::shared_ptr<CFileItem
   return ManageVideoItemArtwork(item, mediaType, "thumb");
 }
 
+namespace
+{
+class CAsyncGetArt : private IRunnable
+{
+public:
+  CAsyncGetArt() = delete;
+  CAsyncGetArt(const VIDEO::IVideoItemArtworkHandler& handler) : m_handler(handler) {}
+
+  bool FetchAllArt()
+  {
+    CGUIDialogBusy::Wait(this, 100, false);
```

Comment:
is the busy handler really needed?

Reply:
How would you solve the problem? In my case it is an nfs call trying to open a file located on a NAS, which is blocked for several seconds until NAS is awake. I thought if the user has to wait for data to be fetched, we should present a spinner?

Do you mean to fill the list with the graphics async? Wow, that would be cool but much effort I guess, because it it currently does not work that way.

Reply:
Yeah I was expecting the last one, sort of lazy loading but I see it's not related (and yeap I perfectly understand it's too much effort). It makes sense to show the spinner as long as it allows the user to understand something is being processed and can be cancelled. Is this operation cancelable at all?

Reply:
No, as I see it, it’s not.

Reply:
If it is not cancelable maybe you can use dialogbusynocancel?

Reply:
I don't see any point why CDialogBusyNoCancel exists, tbh. It is missing `static bool Wait()`, what I need. And looking at the signature of 

`static bool Wait(IRunnable *runnable, unsigned int displaytime, bool allowCancel);`

I seem not to be wrong.